backend/fms_core/services/library.py

```diff
@@ -145,17 +152,18 @@ def convert_library(process, platform, sample_source, container_destination, coo
 
     return library_destination, errors, warnings
 
-def update_library(sample, **kwargs):
+
+def update_library(derived_sample, **kwargs):
     errors = []
     warnings = []
 
-    if sample is None:
+    if derived_sample is None:
         errors.append('Missing sample')
-    elif not sample.is_library:
-        errors.append('Sample {sample.name} is not a library')
+    elif not derived_sample.library:
+        errors.append(f"Sample {derived_sample.biosample.alias} is not a library or a pool of libraries.")
```

Comment:
I think it may be best to leave the alias or name off the error message (user would be confused to see the Alias if it does not match the name.